src/sync/mutex.rs

```diff
@@ -217,3 +217,17 @@ impl<T: Display + ?Sized> Display for MutexGuard<'_, T> {
         (**self).fmt(f)
     }
 }
+
+impl<T> crate::annotations::WithName for &Mutex<T> {
+    fn with_name_and_kind(self, name: Option<&str>, kind: Option<&str>) -> Self {
+        (&self.semaphore).with_name_and_kind(name, kind.or(Some("shuttle::sync::Mutex")));
```

Comment:
Why is this `kind.or`ed and the one below is not?

Reply:
The one below is only forwarding the call to this implementation.